src/main/java/com/falazar/farmupcraft/util/StructureUtils.java

```diff
@@ -0,0 +1,73 @@
+package com.falazar.farmupcraft.util;
+
+import com.mojang.logging.LogUtils;
+import net.minecraft.advancements.critereon.LocationPredicate;
+import net.minecraft.core.BlockPos;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.tags.TagKey;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.level.levelgen.structure.BoundingBox;
+import net.minecraft.world.level.levelgen.structure.Structure;
+import net.minecraft.world.level.levelgen.structure.StructureStart;
+import net.minecraft.world.phys.AABB;
+import org.slf4j.Logger;
+
+import java.util.List;
+
+public class StructureUtils {
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+
+    public static boolean isInStructure(ServerLevel serverLevel, BlockPos pos, ResourceKey<Structure> structure) {
```

Comment:
great!
we can tell if we are in a village with this?
can we get a onenterchunk check for villages?